### PR TITLE
valuta: init at 1.2.0

### PR DESCRIPTION
**Pull request by arthsmn**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/va/valuta/package.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  dontWrapGApps = true;
+
+  preFixup = ''
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+  '';
+
+  meta = with lib; {
+    description = "This is a simple application for converting currencies, with support for various APIs";
+    homepage = "https://github.com/ideveCore/Valuta";
+    license = licenses.gpl3Only;
```
Comment (Aleksanaa) [inline pkgs/by-name/va/valuta/package.nix:64]:
```suggestion
    license = licenses.gpl3Plus;
```

See https://github.com/ideveCore/Valuta?tab=readme-ov-file#license

File: pkgs/by-name/va/valuta/package.nix [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  ];
+
+  dontWrapGApps = true;
+
+  preFixup = ''
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+  '';
+
+  meta = with lib; {
+    description = "This is a simple application for converting currencies, with support for various APIs";
```
Comment (Aleksanaa) [inline pkgs/by-name/va/valuta/package.nix:62]:
```suggestion
    description = "A simple application for converting currencies, with support for various APIs";
```

File: pkgs/by-name/va/valuta/package.nix [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+, libsoup_3
+, libadwaita
+, wrapGAppsHook4
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "valuta";
+  version = "1.2.0";
+
+  format = "other";
```
Comment (Aleksanaa) [inline pkgs/by-name/va/valuta/package.nix:23]:
```suggestion
  pyproject = false;
```

File: pkgs/by-name/va/valuta/package.nix [hunk 4]
```diff
@@ -0,0 +1,69 @@
+
```
Comment (Aleksanaa) [inline pkgs/by-name/va/valuta/package.nix:1]:
```suggestion
```

File: pkgs/by-name/va/valuta/package.nix [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  nativeBuildInputs = [
+    meson
+    ninja
+    cmake
+    pkg-config
+    appstream-glib
+    blueprint-compiler
+    desktop-file-utils
+    wrapGAppsHook4
```
Comment (Aleksanaa) [inline pkgs/by-name/va/valuta/package.nix:39]: Also please avoid double wrapping, see https://github.com/NixOS/nixpkgs/pull/303848

File: pkgs/by-name/va/valuta/package.nix [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    owner = "ideveCore";
+    repo = "Valuta";
+    rev = "v${version}";
+    hash = "sha256-ygroF9hlmsjQTocly7CNL6zQi4N6h8va/ZkJ8w8QHms=";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    cmake
```
Comment (Aleksanaa) [inline pkgs/by-name/va/valuta/package.nix:34]: They are not using cmake?

File: pkgs/by-name/va/valuta/package.nix [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    pkg-config
+    appstream-glib
```
Comment (Aleksanaa) [inline pkgs/by-name/va/valuta/package.nix:36]: Should not be needed

File: pkgs/by-name/va/valuta/package.nix [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+  preFixup = ''
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+  '';
+
+  meta = with lib; {
+    description = "A simple application for converting currencies, with support for various APIs";
+    homepage = "https://github.com/ideveCore/Valuta";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ arthsmn ];
+    mainProgram = "valuta";
```
Comment (marcusramberg) [inline pkgs/by-name/va/valuta/package.nix:62]: Looks like this package creates a binary named `currencyconverter`, not `valuta`?

Comment (arthsmn) [reply pkgs/by-name/va/valuta/package.nix]: Sorry, forgot about this!

**Review by Aleksanaa — APPROVED** [review]
Haven't tested

**Comment by marcusramberg** [comment]
> Haven't tested 

I've confirmed the app starts.

**Pull request merged**